Added Josephus Problem

mShashank2308 opened the pull request.

Idhant-6 commented:
Hi, Can you just provide us with some more details on how does this project works?

mShashank2308 commented:
Josephus problem is a very standard problem in Computer Science and Mathematics, and is asked during interviews. I have added the code which uses recursion. I cannot explain this problem here. For your reference, I have added the [link](https://en.wikipedia.org/wiki/Josephus_problem).
